pandas/_libs/tslibs/conversion.pyx

```diff
@@ -222,7 +222,7 @@ def ensure_datetime64ns(arr: ndarray, copy: bool=True):
         dtype = arr.dtype
         arr = arr.astype(dtype.newbyteorder("<"))
 
-    ivalues = arr.view(np.int64).ravel("K")
+    ivalues = arr.view(np.int64).ravel("C")
 
     result = np.empty(shape, dtype=DT64NS_DTYPE)
```

Comment:
would it make a difference it we constructed this with `result = np.empty_like(arr, dtype=...)` to ensure we keep the same contiguity?

Reply:
Oh yes, fantastic. Thanks very much